src/coreComponents/constitutive/ConstitutivePassThru.hpp

```diff
@@ -52,36 +53,46 @@ template<>
 struct ConstitutivePassThru< SolidBase >
 {
 
+  // NOTE: The switch order here can be fragile if a model derives from another
+  //       model, as the dynamic_cast will also cast to a base version.
+  //       Models should be ordered such that children come before parents.
+  //       For example, DruckerPrager before ElasticIsotropic, DamageVolDev before
+  //       Damage, etc.
+
```

Comment:
That would work better for sure, although I would have a switch/if yard based on `typeid( constitutiveRelation )`. That way you don't need to keep strings up to date.

Reply:
I've added this to the TODO list for a follow-on PR.

Reply:
I just tried out the `typeid` approach and it breaks the `ConstitutivePassThru< SolidBase >` when the model is of type `PoroElastic< SomeTypeDerivedFromSolidBase >`. For example currently `PoroElastic< LinearElasticAnisotropic >` would dispatch as `LinearElasticAnisotropic`, but with `typeid` it won't work at all. Something similar would happen with using `getCatalogName`, but that's even worse because does a template class like `PoroElastic< LinearElasticAnisotropic >` have a catalog name?

We could obviously add in the extra dispatches, which is what's done with the `Damage` classes. I think we should be careful though about adding in extra cases. If all that's needed is for example `LinearElasticAnisotropic` then we don't want to add in ten cases for each of the possible derived types since that will blow up compile time.

Reply:
I think the catalog name is currently `string( "Poro" ) + BASE::m_catalogNameString`

Reply:
Ahh nice. Well I still like the `typeid` approach better, since you'd still have to list out all the types either way.

Reply:
Totally agree.  I'm eager to get this PR finished because it has gotten out of control (my fault) but this is not the last constitutive PR by any means.  I think we'll need 2-3 very focused ones as a follow on.  This issue is at the top of the list.